docs/dependency-injection/dependency-injection.md

```diff
@@ -0,0 +1,206 @@
+# 1. What is Dependency Injection (DI)?
+
+Dependency injection (DI) is a design pattern that provide loosely-coupled, maintainable, and testable code. It improves modularity, maintainability, and testability, and is easily customizable and extensible.
+
+# 2. Service layer in Oryx
+
+In Oryx, dependency injection (DI) serves as the foundation for the business logic. By using DI to provide service layer, Oryx is designed to be easily customizable and extensible on every level, making it an ideal choice for projects with complex or rapidly-evolving requirements.
+
+The goal of service layer in Oryx is to  abstracts all of the system functionality,  including query backend services,  data caching and reloading, state management, and reactivity.
+
+DI is also utilized by many higher-level concepts in Oryx, ranging from adapters, normalizers,  HTTP interceptors, to authentication, product loading, cart management, and checkout process.
+
+## @spryker-oryx/di package
+
+The @spryker-oryx/di library provides dependency injection to Oryx applications, including a variety of useful utilities to streamline working with the DI. This includes:
+
+- TypeScript models definitions (e.g. Providers)
+-  `Injector` - the mplementation of a DI container in TypeScript
+-  Functions used to set up and manage injectors (`createInjector()`, `getInjector()`, etc.)
+-   Utilities that can be used to resolve dependencies (`inject` and `resolve`)
+
+In a typical Oryx application, much of the work related to setting up the DI container is handled automatically by the application orchestrator. However, the @spryker-oryx/di library provides a  set of tools that can be used to further customize and fine-tune the DI as needed.
+
+# 3. Using services
+
+## inject vs resolve
+
+There are two primary methods for injecting services and dependencies into components: `inject` and `resolve`.
```

Comment:
what about:

There are two primary methods for injecting dependencies from the application container:
- `inject` – used inside services
- `resolve` – used inside (web) components